be/src/vec/aggregate_functions/aggregate_function_regr_slope.h

```diff
@@ -0,0 +1,167 @@
+#include "vec/aggregate_functions/aggregate_function.h"
+#include "vec/columns/column.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/common/assert_cast.h"
+#include "vec/core/field.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type_decimal.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/io/io_helper.h"
+namespace doris {
+namespace vectorized {
+class Arena;
+class BufferReadable;
+class BufferWritable;
+template <typename T>
+class ColumnDecimal;
+template <typename>
+class ColumnVector;
+} // namespace vectorized
+} // namespace doris
+
+namespace doris::vectorized {
+
+template <typename T>
+struct AggregateFunctionRegrSlopeData {
+    UInt64 count = 0;
+    double sum_x {};
+    double sum_y {};
+    double sum_of_x_mul_y {};
+    double sum_of_x_squared {};
+
+    void write(BufferWritable& buf) const {
+        write_binary(sum_x, buf);
+        write_binary(sum_y, buf);
+        write_binary(sum_of_x_mul_y, buf);
+        write_binary(sum_of_x_squared, buf);
+        write_binary(count, buf);
+    }
+
+    void read(BufferReadable& buf) {
+        read_binary(sum_x, buf);
+        read_binary(sum_y, buf);
+        read_binary(sum_of_x_mul_y, buf);
+        read_binary(sum_of_x_squared, buf);
+        read_binary(count, buf);
+    }
+
+    void reset() {
+        sum_x = {};
+        sum_y = {};
+        sum_of_x_mul_y = {};
+        sum_of_x_squared = {};
+        count = 0;
+    }
+
+    double get_slope_result() const {
+        double denominator = count * sum_of_x_squared - sum_x * sum_x;
+        if (count == 0 || denominator == 0.0) {
+            return std::numeric_limits<double>::quiet_NaN();
+        }
+        return (count * sum_of_x_mul_y - sum_x * sum_y) / denominator;
+    }
+    
+
+    void merge(const AggregateFunctionRegrSlopeData& rhs) {
+        if (rhs.count == 0) {
+            return;
+        }
+        sum_x += rhs.sum_x;
+        sum_y += rhs.sum_y;
+        sum_of_x_mul_y += rhs.sum_of_x_mul_y;
+        sum_of_x_squared += rhs.sum_of_x_squared;
+        count += rhs.count;
+    }
+
+    void add(const IColumn* column_y, const IColumn* column_x, size_t row_num) {
+        const auto& sources_x = assert_cast<const ColumnVector<T>&, TypeCheckOnRelease::DISABLE>(*column_x);
+        const auto& sources_y = assert_cast<const ColumnVector<T>&, TypeCheckOnRelease::DISABLE>(*column_y);
+        const auto& value_x = sources_x.get_data()[row_num];
+        const auto& value_y = sources_y.get_data()[row_num];
+        sum_x += value_x;
+        sum_y += value_y;
+        sum_of_x_mul_y += value_x * value_y;
+        sum_of_x_squared += value_x * value_x;
+        count += 1;
+    }
+};
+
+template <typename T, typename Data>
+struct RegrSlopeData : AggregateFunctionRegrSlopeData<T> {
+    void insert_result_into(IColumn& to) const {
+        auto& col = assert_cast<ColumnFloat64&, TypeCheckOnRelease::DISABLE>(to);
+        col.get_data().push_back(this->get_slope_result());
```

Comment:
`get_slope_result` would return `Nan` in some situation, but doris does not support Nan, we should get NULL. So result type should be ColumnNullable

Reply:
Does this mean that we have decided to only support Nullable results?

Reply:
@Yoruet  Yes.